tests/devices/test_default_qubit_tracking.py

```diff
@@ -0,0 +1,246 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the tracking capabilities of default qubit.
+"""
+import pytest
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.resource import Resources
+from pennylane.devices import ExecutionConfig
+
+
+class TestTracking:
+    """Testing the tracking capabilities of DefaultQubit."""
+
+    def test_tracker_set_upon_initialization(self):
+        """Test that a new tracker is intialized with each device."""
+        assert qml.device("default.qubit").tracker is not qml.device("default.qubit").tracker
+
+    def test_tracker_not_updated_if_not_active(self):
+        """Test that the tracker is not updated if not active."""
+        dev = qml.device("default.qubit")
+        assert len(dev.tracker.totals) == 0
+
+        dev.execute(qml.tape.QuantumScript())
+        assert len(dev.tracker.totals) == 0
+        assert len(dev.tracker.history) == 0
+
+    def test_tracking_batch(self):
+        """Test that the new default qubit integrates with the tracker."""
+
+        qs = qml.tape.QuantumScript([], [qml.expval(qml.PauliZ(0))])
+
+        dev = qml.device("default.qubit")
+        config = ExecutionConfig(gradient_method="adjoint")
+        with qml.Tracker(dev) as tracker:
+            dev.execute(qs)
+            dev.compute_derivatives(qs, config)
+            dev.execute([qs, qs])  # and a second time
+
+        assert tracker.history == {
+            "batches": [1, 1],
+            "executions": [1, 1, 1],
+            "simulations": [1, 2],
```

Comment:
```
"executions": [1, 1, 1],
"simulations": [1, 2],
```
I'm not sure how to interpret this. Why are executions listed as 3 separate occurrences of 1 each, but simulations are grouped?

Reply:
I rewrote it so that the simulation number is updated at the same time as executions, shots, and resources.  

Now you would get:
```
"executions": [1, 1, 1],
"simulations": [1, 1, 1],
```

Does this make more sense?

Reply:
To me that feels more clear, thanks!